File: lógica/JS/vetor/alturas.js
// Problema "alturas" 
// Fazer um programa para ler nome, idade e altura de N pessoas. Depois, mostrar na tela a altura média das pessoas, e mostrar também a porcentagem de pessoas com menos de 16 anos, bem como os nomes dessas pessoas caso houver.

var N, media, soma, porc, cont;
var nome = [];
var altura = [];
var idade = [];

N = parseInt(prompt('Quantas pessoas serão digitadas ?'));

for(let i = 0; i < N; i++) {
    nome = prompt('Nome:');
    idade[i] = parseInt(prompt('Idade:'));
    altura[i] = parseFloat(prompt('Altura:'));
}

soma = 0;
for(let i = 0; i < altura.length; i++) {
    soma = soma + altura[i]; 
}
media = soma / N;
console.log('Altura média: ', media.toFixed(2));

cont = 0;
for(let i = 0; i < idade.length; i++) {
    if(idade[i] < 16) {
        cont = cont + 1;
    }
}
porc = cont * 100 / N;
console.log('Pessoas com menos de 16 anos: ', porc.toFixed(2), '%')

for(let i = 0; i < idade.length; i++) {
    if(idade[i] < 16) {
        console.log(nome);
    }
}

File: lógica/Python/vetor/alturas.py
# Problema "alturas" 
# Fazer um programa para ler nome, idade e altura de N pessoas. Depois, mostrar na tela a altura  média das pessoas, e mostrar também a porcentagem de pessoas com menos de 16 anos, bem como os nomes dessas pessoas caso houver.

n: int = int(input("Quantas pessoas serao digitadas ? ")) 

nomes: str = [0 for x in range(n)]  
idades: int = [0 for x in range(n)]
alturas: float = [0 for x in range(n)]

for i in range(n):
    print(f"Dados da {i+1}a pessoa")
    nomes[i] = str(input('Nome: '))
    idades[i] = int(input('Idade: '))
    alturas[i] = float(input('Altura: '))

alturaTotal = 0
menor = 0
for i in range(n):
    if idades[i] < 16:
        menor = menor + 1
        alturaTotal = alturaTotal + alturas[i]

alturaMedia = alturaTotal/menor
percentual = (float(menor)/n) * 100.0

print(f"\nAltura media = {alturaMedia:.2f}")
print(f"Pessoas com menos de 16 anos: {percentual:.1f}%")

for i in range(n):
    if idades[i] < 16:
        print(nomes[i])